torch/optim/sgd.py

```diff
@@ -354,6 +354,8 @@ def _single_tensor_sgd(
         if isinstance(lr, Tensor) and lr.requires_grad:
             param.addcmul_(grad, lr, value=-1)
         else:
+            # CPU scalar tensors w/out grad works but isn't supported in typehints
+            lr = cast(float, lr)
```

Comment:
It should almost always already be a float--we don't want to do this in the sgd code, but we should enforce that it's always a float in the tests

Reply:
Okay! I added it bc the CI failed, which I think is because functional uses really strict typing. What do u think we should do about the strict typing?

Reply:
I was thinking there's a chance we could do just if `isinstance(lr, Tensor)` because the error on mps was because I hadn't allowed the function to even take in cpu_scalars before the previous PR we worked on was merged. I think the test we added that tests the scalar tested all the devices and it seemed to work on all of them, but I could be terribly mistaken. 

I also am kinda concerned that any tensor LRs would go into the addcmul and there might be some behavior supported w/ add that's not with addcmul, but it would fix this typing issue

Reply:
What happens when you remove the type hint at the definition of this function?

Generally, jit script is too rigid with this..we shouldn't change our code semantics to make typing for jit script happy. Also, the addcmul path on MPS probably would not work even with your change because your previous PR is CUDA only.